showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITOtelMetrics.java

```diff
@@ -0,0 +1,554 @@
+                EchoSettings.defaultHttpJsonTransportProviderBuilder()
+                    .setHttpTransport(
+                        new NetHttpTransport.Builder().doNotValidateCertificate().build())
+                    .setEndpoint("http://localhost:7469")
+                    .build())
+            .build();
+    EchoClient client = EchoClient.create(httpJsonEchoSettings);
+
+    EchoRequest requestWithError =
+        EchoRequest.newBuilder()
+            .setError(Status.newBuilder().setCode(Code.PERMISSION_DENIED.ordinal()).build())
+            .build();
+
+    client.echoCallable().futureCall(requestWithError).isDone();
+
+    // wait for the metrics to get uploaded
+    Thread.sleep(5000);
+
+    String filePath = "../directory5/testHttpJson_OperationFailed_metrics.txt";
+    File file = new File(filePath);
+    Truth.assertThat(file.exists()).isTrue();
+    Truth.assertThat(file.length() > 0).isTrue();
+
+    client.close();
+    client.awaitTermination(TestClientInitializer.AWAIT_TERMINATION_SECONDS, TimeUnit.SECONDS);
+  }
+
+  @Test
+  public void testGrpc_OperationFailed_recordsMetrics() throws Exception {
+
+    // initialize the otel-collector
+    setupOtelCollector(
+        "../scripts/start_otelcol.sh",
+        "../directory6",
+        "../opentelemetry-helper/configs/testGrpc_OperationFailed.yaml");
+
+    EchoSettings grpcEchoSettings =
+        EchoSettings.newBuilder()
+            .setCredentialsProvider(NoCredentialsProvider.create())
+            .setTracerFactory(createOpenTelemetryTracerFactory("4322"))
+            .setTransportChannelProvider(
+                EchoSettings.defaultGrpcTransportProviderBuilder()
+                    .setChannelConfigurator(ManagedChannelBuilder::usePlaintext)
+                    .build())
+            .setEndpoint("localhost:7469")
+            .build();
+
+    EchoClient client = EchoClient.create(grpcEchoSettings);
+
+    EchoRequest requestWithError =
+        EchoRequest.newBuilder()
+            .setError(Status.newBuilder().setCode(Code.UNAUTHENTICATED.ordinal()).build())
+            .build();
+
+    client.echoCallable().futureCall(requestWithError).isDone();
+
+    // wait for the metrics to get uploaded
+    Thread.sleep(5000);
+
+    String filePath = "../directory6/testGrpc_OperationFailed_metrics.txt";
+    // 1. Check if the log file exists
+    File file = new File(filePath);
+    Truth.assertThat(file.exists()).isTrue();
+    Truth.assertThat(file.length() > 0).isTrue();
+
+    client.close();
+    client.awaitTermination(TestClientInitializer.AWAIT_TERMINATION_SECONDS, TimeUnit.SECONDS);
+  }
+
+  @Test
+  public void testGrpc_attemptFailedRetriesExhausted_recordsMetrics() throws Exception {
+
+    setupOtelCollector(
+        "../scripts/start_otelcol.sh",
+        "../directory7",
+        "../opentelemetry-helper/configs/testGrpc_attemptFailedRetriesExhausted.yaml");
+
+    RetrySettings retrySettings = RetrySettings.newBuilder().setMaxAttempts(2).build();
```

Comment:
qq, some tests use 2, 3, or 5 max attempts? Is there a reason for the different numbers?

Reply:
No reason for these different numbers. The count appears in the metrics collected, so I will use these numbers in the verification.